### PR TITLE
Add release note for CoreDNS

### PR DESCRIPTION
**Pull request by Cynerva**

### CODE REVIEW

File: pages/k8s/release-notes.md [hunk 1]
```diff
@@ -60,4 +82,4 @@ Please see [this page][historic] for release notes of earlier versions.
 [haoverview]:
 [metallb-docs]:
 [hacluster-docs]:
-
+[dns-provider config description]: https://github.com/juju-solutions/kubernetes/blob/5f4868af82705a0636680a38d7f3ea760d35dadb/cluster/juju/layers/kubernetes-master/config.yaml#L58-L67
```
Comment (Cynerva) [inline pages/k8s/release-notes.md]: This config description should probably be brought into kubernetes-docs somewhere, but I wasn't quite sure where it should go. New page probably, maybe **Operations -> Addons** or **Reference -> Charm configuration**?

**Comment by Cynerva** [comment]
The dns-provider config description link isn't rendering right. It looks right to me, so I'm guessing it's just because of the other links in the doc that are missing URLs. Let me know if I should look into it further.

File: pages/k8s/release-notes.md (hunk 1 above)
Comment (evilnick) [reply pages/k8s/release-notes.md]: It should. A charm-configuration set of pages is planned, but for now I think it makes perfect sense for this to go in the 'ugrade-notes' pages, linked to from here. I will add those changes to this PR so you can see.

**Comment by evilnick** [comment]
@Cynerva I moved the link to the upgrade-motes page. seems to be working fine in the preview now.
Let me know if this re-org makes sense to you

**Comment by Cynerva** [comment]
@evilnick Yep, that makes a lot of sense. LGTM, thanks!

File: pages/k8s/upgrade-notes.md [hunk 2]
```diff
@@ -13,7 +13,40 @@ layout: [base, ubuntu-com]
...
+any of the intervening steps.
+
+<a  id="1.14"> </a>
+
+## Upgrading to 1.14
+
+This upgrade includes support for **CoreDNS**. All new deployments of **CDK**
+with Kubernetes 1.14 will install **CoreDNS** by default instead of **KubeDNS**.
+
+Existing deployments that are upgraded to **CDK** and Kubernetes 1.14 will
```
Comment (tvansteenburgh) [inline pages/k8s/upgrade-notes.md]: s/CDK and Kubernetes 1.14/CDK 1.14/

Comment (Cynerva) [reply pages/k8s/upgrade-notes.md]: Fixed, thanks

Comment (evilnick) [reply pages/k8s/upgrade-notes.md]: you missed one, but I got it, and as a bonus I found a 'that' which should have been a 'which'

Comment (tvansteenburgh) [reply pages/k8s/upgrade-notes.md]: :cake:

Comment (Cynerva) [reply pages/k8s/upgrade-notes.md]: Oops. Good catches, thanks!

**Pull request merged**